File: library/src/main/java/cn/wandersnail/adapter/tree/OnExpandableItemLongClickListener.java
package cn.wandersnail.adapter.tree;

import android.view.View;
import android.widget.AdapterView;

/**
 * 可展开的条目长按事件监听
 * <p>
 * date: 2019/8/21 22:05
 * author: zengfansheng
 */
public interface OnExpandableItemLongClickListener<T extends Node<T>> {
    /**
     * @param node     被长按条目的数据
     * @param parent   被长按的父容器
     * @param view     被长按的View
     * @param position 被长按条目所在整个数据集合中的索引
     */
    void onExpandableItemLongClick(T node, AdapterView<?> parent, View view, int position);
}

File: library/src/main/java/com/snail/treeadapter/OnExpandableItemLongClickListener.kt
package com.snail.treeadapter

import android.view.View
import android.widget.AdapterView

/**
 * 时间: 2017/12/5 11:20
 * 作者: zengfansheng
 * 功能:
 */

interface OnExpandableItemLongClickListener<T : Node<T>> {
    fun onExpandableItemLongClick(node: T, parent: AdapterView<*>, view: View, position: Int)
}
